addon/background.js

```diff
@@ -52,44 +53,46 @@ function setInstalledTxpAddons() {
 }
 
 async function setup() {
-  const { clientUUID } = await storage.get("clientUUID");
-  if (!clientUUID) {
-    await storage.set({ clientUUID: uuidv4() });
-  }
   setupEnvironment();
   setupBrowserAction();
   await fetchResources();
-  setDailyPing();
+  await setDailyPing();
+}
+
+async function sendPingOnAlarm(alarmInfo) {
+  if (alarmInfo.name === "daily-ping") {
+    const { lastPing } = await storage.get("lastPing");

```

Comment:
I was thinking more along the lines of `storage.get({ lastPing: Date.now() - ONE_DAY })`

Reply:
I don't think this would ever return successfully :thinking: